### PR TITLE
First Implementation of the OTP Read/Write function

### PR DESCRIPTION
**Pull request by mirkomatonti**
Hi,

I need to use the OTP area so I decided to provide an implementation for those boards that treat the OTP area as another FLASH region. 

Here is the basic structure of my implementation:
--area=otp read <output.bin> <len> //Read from the OTP start address to len
--area=otp write <input.bin> <add> //Write from a specific OTP address

Please note that I have only been able to test my implementation on the l452 and l4a6 boards as they are the only ones I have access to.

[Transcript](https://gist.github.com/mirkomatonti/a7a784d4d460f95e0cf2da4b5b31292d)

[Link to discussion](https://github.com/stlink-org/stlink/issues/1352)

### CODE REVIEW

**Comment by mirkomatonti** [comment]
Forgot to include a file :(

**Review by Nightwalker-87 — APPROVED** [review]

**Comment by Nightwalker-87** [comment]
@Ant-ON: Waiting for second review now.

File: src/stlink-lib/common_flash.c
Comment (Ant-ON) [inline src/stlink-lib/common_flash.c:1315]: The condition should look like:
```c
if(addr >= sl->otp_base && addr < sl->otp_base + sl->otp_size) {
```

Comment (Nightwalker-87) [reply src/stlink-lib/common_flash.c:1315]: indeed...

Comment (mirkomatonti) [reply src/stlink-lib/common_flash.c:1315]: Yeah indeed. Will fix that tomorrow 👍

**Review by Nightwalker-87 — APPROVED** [review]

**Pull request merged**
